type-definitions/ts-tests/map.ts

```diff
@@ -80,6 +115,28 @@ import { Map, List } from 'immutable';
 
   // $ExpectType Map<number, string | number>
   Map<number, number | string>().set(0, 'a');
+
+  // $ExpectError
+  Map({ a: 1 }).set('b', 'b');
```

Comment:
I'm surprised by this, I'd expect either `MapFromObject<{ a: 1, b: 'b' }>` or `Map<string, number | string>` as a result?

Reply:
See comment above about the consistency with plain objects in TS